bgpd/bgp_route.c

```diff
@@ -14216,6 +14216,21 @@ static int bgp_clear_damp_route(struct vty *vty, const char *view_name,
 				while (pi) {
 					if (pi->extra && pi->extra->damp_info) {
 						pi_temp = pi->next;
+						struct bgp_damp_info *bdi =
+							pi->extra->damp_info;
+						if (bdi->lastrecord
```

Comment:
Shouldn't we check for pi->extra before dereferencing to damp_info in advance?

Reply:
sorry, I did not get your question. We are already checking it in line number 14217. if(pi->extra && pi->extra->damp_info)

Reply:
Beat me.